### PR TITLE
mgr/cephadm: Add config() to CephadmService

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Salvaged from #39223

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/services/cephadmservice.py [hunk 1]
```diff
@@ -525,7 +532,7 @@ def mgr_map_has_standby(self) -> bool:
 class MdsService(CephService):
     TYPE = 'mds'
 
-    def config(self, spec: ServiceSpec) -> None:
+    def config(self, spec: ServiceSpec, daemon_id: str) -> None:
```
Comment (jmolmo) [inline src/pybind/mgr/cephadm/services/cephadmservice.py:535]: #type: ignore?

**Review by jmolmo — APPROVED** [review]
LGTM

File: src/pybind/mgr/cephadm/services/cephadmservice.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/services/cephadmservice.py:535]:
```python
class Spec: ...

class RGWSpec(Spec): ...

class CephService:
    def config(self, spec: Spec): ...

class RGWService(CephService):
    def config(self, spec: RGWSpec): ...
```

```
file.py:9: error: Argument 1 of "config" is incompatible with supertype "CephService"; supertype defines the argument type as "Spec"
file.py:9: note: This violates the Liskov substitution principle
file.py:9: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
Found 1 error in 1 file (checked 1 source file)
```

This is only required in derived classes that demand a subclass of ServicSpec

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
```
flake8 run-test-pre: PYTHONHASHSEED='2707289287'
flake8 run-test: commands[0] | flake8 --config=tox.ini cephadm orchestrator
cephadm/serve.py:5:1: F401 'typing.Callable' imported but unused
cephadm/serve.py:17:1: F401 'ceph.deployment.service_spec.RGWSpec' imported but unused

**Comment by sebastian-philipp** [comment]
* https://pulpito.ceph.com/swagner-2021-02-10_11:41:39-rados:cephadm-wip-swagner-testing-2021-02-09-1126-distro-basic-smithi/
* https://pulpito.ceph.com/swagner-2021-02-11_10:31:33-rados:cephadm-wip-swagner-testing-2021-02-09-1126-distro-basic-smithi/

Failures:

* https://tracker.ceph.com/issues/49259
* e2e test

**Comment by sebastian-philipp** [comment]
changelog: fixed flake8

**Pull request merged**